src/lib/evil/evil_private.h

```diff
@@ -56,9 +56,6 @@ extern "C" {
 #define sigsetjmp(Env, Save) setjmp(Env)
 
 #ifdef _MSC_VER
-#ifndef WIN32_LEAN_AND_MEAN
-# define WIN32_LEAN_AND_MEAN
```

Comment:
Why removing this?

Reply:
It is not necessary as it simply prevents nothing: some file outside of evil could be including `windows.h` without including `evil_private.h` therefore not using this definition. Now those that are also including <evil_private.h>, that is correctly using `WIN32_LEAN_AND_MEAN` to prevent <windows.h> from including `<winsock.h>` instead of `<winsock2.h>`.

I think that if `WIN32_LEAN_AND_MEAN` should be globally defined it should be defined in `meson.build`.

Reply:
okay, as [basetsd;h](https://github.com/tpn/winsdk-10/blob/master/Include/10.0.10240.0/shared/basetsd.h) doesn't include any other windows header it should be ok.